.github/workflows/ci.yml

```
@@ -1,292 +1,350 @@
+    - name: Upload Bandit report
+      uses: actions/upload-artifact@v3
+      if: always()
+      with:
+        name: bandit-report
+        path: bandit-report.json
+
+    - name: Run Safety security check
+      run: safety check --json --output safety-report.json || true
+
+    - name: Upload Safety report
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.8)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)